modules/web-console/web-console-common/src/main/java/org/apache/ignite/console/event/Event.java

```diff
@@ -0,0 +1,68 @@
+package org.apache.ignite.console.event;
+
+import org.springframework.context.ApplicationEvent;
+
+/**
+ * Generic application event with payload
+ */
+public class Event<T> extends ApplicationEvent {
+    /** Type. */
+    private Type type;
+
+    /** Payload. */
+    private T payload;
+
+    /**
+     * @param type Type.
+     * @param payload Payload.
+     */
+    public Event(Type type, T payload) {
+        super(new Object());
+
+        this.type = type;
+        this.payload = payload;
+    }
+
+    /**
+     * @return Type.
+     */
+    public Type getType() {
+        return type;
+    }
+
+    /**
+     * @return Payload.
+     */
+    public T getPayload() {
+        return payload;
+    }
+
+    /**
+     * Event type
+     */
+    public enum Type {
```

Comment:
I think we should make it a standalone enum.
https://stackoverflow.com/questions/4149426/never-use-public-nested-enums

Reply:
Done